Runtime/DialogueViewBase.cs

```diff
@@ -62,104 +40,65 @@ public virtual void DialogueStarted()
-
-        private IEnumerator RunLineWithCallback (LocalizedLine dialogueLine, Action onDialogueLineFinished) {
-            dialogueRunnerCurrentLine = onDialogueLineFinished.Target as DialogueRunner;
-            dialogueLineStatus = DialogueLineStatus.Running;
-
-            yield return StartCoroutine(RunLine(dialogueLine));
-
-            dialogueLineStatus = DialogueLineStatus.Finished;
-            onDialogueLineFinished();
-        }
+        /// <param name="dialogueLine">The content of the line that should
+        /// be presented to the user.</param>
+        /// <param name="onDialogueLineFinished">The method that should be
+        /// called after the line has been finished.</param>
+        /// <returns>Returns <see
+        /// cref="Dialogue.HandlerExecutionType.PauseExecution"/> if
+        /// dialogue should wait until the completion handler is called
+        /// before continuing execution;
+        /// <see cref="Dialogue.HandlerExecutionType.ContinueExecution"/>
+        /// if dialogue should immediately continue running after calling
+        /// this method.</returns>
+        /// FIXME: If this method is expected to be called only from the
+        /// DialogueRunner then this should be converted into a coroutine
+        /// and merged with RunLineWithCallback();
+        public abstract void RunLine(LocalizedLine dialogueLine, Action onDialogueLineFinished);
 
         /// <summary>
-        /// Finish presenting the current <see cref="LocalizedLine"/>.
+        /// Called by the DialogueRunner to indicate that the line that
+        /// this view is delivering has changed state.
         /// </summary>
         /// <remarks>
-        /// Called if the <see cref="DialogueRunner"/> has received the user's request on one of the 
-        /// <see cref="DialogueRunner.dialogueViews"/> classes derived from <see cref="DialogueViewBase"/> to 
-        /// finish presenting the current line.
+        /// Subclasses of <see cref="DialogueViewBase"/> should override
+        /// this method to be notified when a line has become interrupted,
+        /// and when the line has finished being delivered by all views.
+        ///
+        /// The default implementation does nothing.
         /// </remarks>
-        internal void FinishRunningCurrentLine() {
-            dialogueLineStatus = DialogueLineStatus.Interrupted;
-            FinishCurrentLine();
+        /// <param name="dialogueLine">The <see cref="LocalizedLine"/> that
+        /// has changed state.</param>
+        /// <param name="previousStatus">The status that the line used to
+        /// have.</param>
+        /// <param name="newStatus">The status that the line now
+        /// has.</param>
+        /// <seealso cref="LineStatus"/>
+        public virtual void OnLineStatusChanged(LocalizedLine dialogueLine, LineStatus previousStatus, LineStatus newStatus) {
+            // default implementation is a no-op
         }
```

Comment:
I'm not sure if it's necessary to have the LocalizedLine and the previous LineStatus as parameters. It's not being used in any of its derived classes. Passing the previous LineStatus from the DialogueRunner also seems to work against this PR's intent of moving ownership of the LineStatus to the LocalizedLine itself.

Reply:
Agreed. I've made `OnLineStatusChanged` take a single parameter, the `LocalizedLine`.